### ISSUE TITLE
Regression: Senza delete no longer removes outer R53 entry when no stacks remain

### ISSUE THREAD

**Issue by beverage** [zalando-stups/senza#268]
(Could alternatively be named 'senza create attempts to recreate outer R53 entry when one already exists')

This causes subsequent senza create to fail.  It seems that when no stacks are detected, it tries to create the entry again, but since it still exists CloudFormation fails the operation.

For example, for a single stack service, the following R53 entries will exist:

1. service-name-blah.domain
2. service-name-blah-stackid.domain

After senza delete:

1. service-name-blah.domain

When more than one stack id exist, it seems senza create will not attempt to recreate service-name-blah.domain.  When no stack id suffixed entries exist, it does try to recreate service-name-blah.domain.

Perhaps the assumption was that all service will have multiple stacks with traffic shifting as the method of activating the new service?

**Comment by hjacobs** [comment]
I think this behavior is due to how CloudFormation works:

* Senza `create` creates a new CloudFormation (CF) stack with VersionDomain and MainDomain DNS records
  * MainDomain has weight zero for this stack version and is a "managed by CF" resource
* Senza `traffic` will change the weight of the MainDomain record without changing the CF template
  * MainDomain is therefore no longer exactly the same resource as originally created by CF
* Senza `delete` will delete the CF stack
  * MainDomain will not be deleted by CF as the resource no longer matches the original "CF managed" resource

So basically we are mixing CloudFormation and manual changes (directly using Route 53 API).

**Comment by hjacobs** [comment]
That's actually a duplicate of #184.
